plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/UpgradeRuntimeQuickFixTest.java

```diff
@@ -61,10 +56,29 @@
       JavaFacet.VERSION_1_7, WebFacetUtils.WEB_25, AppEngineStandardFacet.JRE7);
 
   @Test
-  public void testApply() throws CoreException {
+  public void testConvertJava7() throws CoreException {
+    String appengineWebAppJava7 =
+        "<appengine-web-app xmlns='http://appengine.google.com/ns/1.0'>"
+        + "<runtime>java7</runtime>"
+        + "</appengine-web-app>";
+    
+    checkUpgrade(appengineWebAppJava7);
+  }
+  
+  @Test
+  public void testAddMissingRuntime() throws CoreException {
+    String appengineWebAppJava7 =
+        "<appengine-web-app xmlns='http://appengine.google.com/ns/1.0'>"
```

Comment:
hit: I think you can close an element in itself, like `<element/>`?

Reply:
possible, not required